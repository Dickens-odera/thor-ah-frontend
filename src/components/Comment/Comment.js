import React from 'react';
import PropTypes from 'prop-types';
import styles from './comment.module.scss';
// images
import likeIcon from "../../assets/up.png";
import dislikeIcon from "../../assets/down.png";
// import profileImage from "../../assets/Ellipse.png";

const Comment = ({ comment }) => (
  <div className={`card p-3 ${styles.comment}`}>
    <div className="d-flex ">
      <img className={styles.commenterImg} src={`http://i.pravatar.cc/150?u=${comment.commenter.firstName}`} alt="profile" />
      <span className="font-weight-bold text-secondary p-2">
        {comment.commenter.firstName}
      </span>
    </div>
    <p className="text-secondary text-left my-4">
      {comment.body}
    </p>
    <div className="d-flex">
      <div className="px-2">
        <img src={likeIcon} alt="icon" />
        <span>{comment.likesCount}</span>
      </div>
      <div className="px-2">
        <img src={dislikeIcon} alt="icon" />
        <span>{comment.dislikesCount}</span>
      </div>
    </div>
  </div>
);

Comment.propTypes = {
  comment: PropTypes.object.isRequired,
}

export default Comment;
