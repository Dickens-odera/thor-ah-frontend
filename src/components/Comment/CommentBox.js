import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './comment.module.scss';
import createComment from '../../actions/comments';

class CommentBox extends Component {
  constructor(props) {
    super(props)
    this.state = {
      comment: '',
    };

    this.submitComment = this.submitComment.bind(this);
    this.handleTextAreaChange = this.handleTextAreaChange.bind(this);
  }

  handleTextAreaChange(event) {
    this.setState({
      comment: event.target.value
    });
  }

  submitComment(event) {
    event.preventDefault();
    const { comment } = this.state;
    if (comment.length < 1) return;

    const articleSlug = window.location.pathname.split('/article/')[1];
    const { handleCommentSubmit } = this.props;
    handleCommentSubmit(comment, articleSlug);
    this.setState({
      comment: ''
    });
  }

  render() {
    const { comment } = this.state;
    return (
      <form onSubmit = {this.submitComment} >
        <div className={styles.field}>
          <div className="control">
            <textarea
              className="form-control p-3"
              id="comment"
              placeholder="Join the conversation..."
              rows={3}
              value={comment}
              onChange={this.handleTextAreaChange}
            />
          </div>
          {/* display submit button only as soon as user starts typing */}
          {comment ? <button className={`btn btn-success ${styles.submit}`} type="submit">submit</button> : ''}
        </div>
      </form>
    );
  }
}

CommentBox.propTypes = {
  handleCommentSubmit: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  handleCommentSubmit(comment, slug) {
    dispatch(createComment(comment, slug));
  }
});

export default connect(null, mapDispatchToProps)(CommentBox);
