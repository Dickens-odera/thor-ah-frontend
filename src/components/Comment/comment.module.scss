form {
  border: 1px solid rgba(0, 0, 0, .125);
  border-radius: 5px;
}


.field {
  padding: 10px 15px 20px;
  display: flex;
  flex-direction: column;

    textarea {
      background: #FCFCFC;
    }
}

.submit {
  margin-top: 10px;
  align-self: flex-end;
}

.comment {
  margin-bottom: 20px;
}
