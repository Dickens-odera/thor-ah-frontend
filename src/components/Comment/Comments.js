import React from 'react';
import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies
import Comment from './Comment';

const Comments = ({ comments, currentArticleTitle }) => (
  <React.Fragment>
    <p className="text-secondary text-left py-3">
      {currentArticleTitle ? `Thoughts on "${currentArticleTitle}"` : ''}
    </p>
    <section>
      {comments && comments.map(comment => <Comment key={comment.id} comment={comment} />)}
    </section>
  </React.Fragment>
);

Comments.propTypes = {
  comments: PropTypes.array.isRequired,
  currentArticleTitle: PropTypes.string.isRequired,
}

export default Comments;
