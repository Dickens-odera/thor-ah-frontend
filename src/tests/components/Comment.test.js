import React from "react";
import toJson from 'enzyme-to-json';
import { shallow } from "enzyme";
import Comment from "../../components/Comment/Comment";

describe("Comment Component", () => {
  test("renders the Comment Component", () => {
    const comment = {
      commenter: {
        firstName: 'John'
      }
    }
    const wrapper = shallow(<Comment comment={comment} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
