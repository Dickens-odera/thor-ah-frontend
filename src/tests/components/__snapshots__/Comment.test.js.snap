// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component renders the Comment Component 1`] = `
<div
  className="card p-3 comment"
>
  <div
    className="d-flex "
  >
    <img
      alt="profile"
      className="commenterImg"
      src="http://i.pravatar.cc/150?u=John"
    />
    <div>
      <span
        className="font-weight-bold text-secondary p-2"
      >
        John
      </span>
      <br />
      <span
        className="timestamp"
      >
        almost NaN years
         ago
      </span>
    </div>
  </div>
  <p
    className="text-secondary text-left my-4"
  />
  <div
    className="d-flex"
  >
    <div
      className="px-2"
    >
      <img
        alt="icon"
        src={Object {}}
      />
      <span>
        0
      </span>
    </div>
    <div
      className="px-2"
    >
      <img
        alt="icon"
        src={Object {}}
      />
      <span>
        0
      </span>
    </div>
  </div>
</div>
`;
